### ISSUE TITLE
Icingaweb2 does not work with Basic Auth

### ISSUE THREAD

**Issue by mj84** [HenriWahl/Nagstamon#266]
Hi there,

as of now, Nagstamon 2.0 always assumes form-based authentication for Icingaweb2 servers.
However, authenticating via Basic or Digest authentication is also possible, but Nagstamon crashes when connecting to such an Icingaweb2 instance.

Therefor, a new option for Cookie- / form-based authentication for Icingaweb2 should be introduced.

Regards,
Markus

**Comment by HenriWahl** [comment]
Can you tell me what settings are needed at the server side to get basic auth for IcingaWeb2, so I can tr to reproduce this?

**Comment by mj84** [comment]
I have configured external (webserver) authentication within /etc/icingaweb2/authentication.ini:

[ExternalAuth]
strip_username_regexp = ""
backend = "external"

In my Apache VHost, i have configured basic authentication as such:
  <Location /icingaweb2/>
    AuthName "Icingaweb2"
    AuthType Basic
    AuthUserFile /etc/icingaweb2/htpasswd
    Require valid-user
  </Location>

I was able to make Nagstamon work with my setup by changing Nagstamon/Servers/IcingaWeb2.py.
I moved lines 89 - 101 into an if condition which never comes true (if self.authentication == 'cookie').
This way, the basic and digest authentication methods of the GenericServer class are inherited and it works for me now.
